Resources/Private/JavaScript/Host/Containers/RightSideBar/Inspector/EditorEnvelope/index.js

```diff
@@ -119,26 +57,17 @@ export default class EditorEnvelope extends Component {
             node: node.toJS(),
             value: transientValue ? transientValue.value : sourceValue,
             propertyName: id,
-            identifier: this.generateIdentifier(),
             options
         };
     }
 
     renderEditorComponent() {
-        const {loadingState} = this.state;
-
-        if (loadingState === LOAD_ERROR) {
-            return (<div>An error occurred</div>);
-        }
-
-        if (loadingState === LOAD_PENDING) {
-            return (<div>Loading...</div>);
-        }
 
-        const {EditorComponent} = this.state;
-        const {transient, id, commit} = this.props;
+        const {transient, id, commit, editor} = this.props;
+        const editorDefinition = registry.inspector.editors.get(editor);
 
-        if (EditorComponent) {
+        if (editorDefinition && editorDefinition.component) {
+            const EditorComponent = editorDefinition && editorDefinition.component;
```

Comment:
nitpick: probably no need for `editorDefinition &&` check here, as it's been checked one line above.